Signal/src/ViewControllers/ConversationView/ConversationViewItem.m

```diff
@@ -143,14 +143,24 @@ - (void)updateAuthorConversationColorNameWithTransaction:(YapDatabaseReadTransac
 {
     OWSAssertDebug(transaction);
 
-    if (self.interaction.interactionType != OWSInteractionType_IncomingMessage) {
-        _authorConversationColorName = nil;
-        return;
+    switch (self.interaction.interactionType) {
```

Comment:
Adding a new "transient" cell type turned out to be incredibly easy.  I had to:

* Add a new interaction type.
* Add a new "stub" `TSInteraction` subclass.  This is regrettable, but it seemed simpler than modifying view items to only sometimes have a `TSInteraction`.
* The changes you see in this file.

I've very optimistic about converting unread indicators, date breaks, etc. to transient view items.

Reply:
🎉